### PR TITLE
🎨 Use `tools_metadata` as a submodule in `resources/flutter`

### PR DESCRIPTION
**Pull request by AlexV525**
https://github.com/flutter/tools_metadata is now configured as a submodule under `resources/flutter` instead of copy it manually.

The PR is huge because it deletes copied files. Be careful when reviewing the "Files changed" tab. 😃

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by AlexV525** [comment]
@helin24 Could you help me verify the issue of the Kokoro build? I cannot access it since it looks like an internal infrastructure.

File: tool/kokoro/setup.sh [hunk 1]
```diff
@@ -15,7 +15,6 @@ setup() {
 
   export JAVA_HOME=`pwd`/../jdk-17.0.4.1.jdk/Contents/Home
   export PATH=$PATH:$JAVA_HOME/bin
-  export JAVA_OPTS=" -Djava.net.preferIPv4Stack=false -Djava.net.preferIPv6Addresses=true"
```
Comment (AlexV525) [inline tool/kokoro/setup.sh:18]: Removing this helps Kokoro to build successfully.

**Comment by AlexV525** [comment]
Updating docs references since `flutter.dev` just went down for a few minutes and exposed the issue.

**Comment by AlexV525** [comment]
cc @jwren @helin24 could you review this?

**Comment by AlexV525** [comment]
Also @DanTup

**Comment by AlexV525** [comment]
> One question I'd have is about how this impacts the production builds of the plugin - will it automatically include these files in their new location (or is there some list somewhere of what gets included in the shipped plugin)? If it's automatically all included, will it include the whole tools_metadata repo (including things like the `tool` folder)?

| Previous | After |
|----|----|
| `tools_metadata/resources` | `tools_metadata` |

The `resources/flutter` will include the entire `tools_metadata` starting from this change. I didn't dig into whether irrelevant files (`resources/flutter/!resources`) could affect production builds though.

**Comment by jwren** [comment]
Thanks @AlexV525 for this PR.  With less individuals working on the Flutter Plugin, and with many recent issues with the Flutter IJ build scripts, I have not had the time to verify that these changes won't have any negative consequences with the Gradle scripts, or other scripts we have that help maintain the Flutter Plugin for IntelliJ.

Leaving this PR open for the time being, thank you again for putting it together.

**Comment by AlexV525** [comment]
@jwren Thanks!

> With less individuals working on the Flutter Plugin, and with many recent issues with the Flutter IJ build scripts, I have not had the time to verify that these changes won't have any negative consequences with the Gradle scripts, or other scripts we have that help maintain the Flutter Plugin for IntelliJ.

I'd willing to help to verify if it works well if you can somehow provide any suggestions, and I'm fine to wait until you can figure out all fixes to build scripts. You can let me know once they are good to go.

**Comment by jwren** [comment]
Thank you for creating this PR, but for the time being I am closing it.  PRs can't remain open for this long against the Flutter GitHub repo and with other concerns around the Flutter Plugin for IntelliJ I won't have the cycles to properly review and land this change.

**Pull request closed without merging**